### PR TITLE
Support Phoenix 1.4

### PR DESCRIPTION
**Pull request by scarfacedeb**
I updated deps to support Phoenix 1.4.0-rc and dropped Cowboy, because I don't think it's needed here.

Also I deleted `phoenix.*` tasks, because phoenix 1.4 removed them too.

### CODE REVIEW

**Review by doomspork — APPROVED** [review]
Thank you so much @scarfacedeb!!  This is really great.  I'm going to give @Rakoth a chance to take a peek before I merge 😁

**Comment by mschae** [comment]
Any news on this? Phoenix 1.4 was since released (would be great it @scarfacedeb could update the PR accordingly). Would be great to get this merged and released, soon. If there's any help we can provide please let me know.

**Comment by scarfacedeb** [comment]
@mschae @doomspork I updated `mix.exs` to use `phoenix 1.4` and templates to use `Routes.` module for path helpers.

I think it should be ready to merge.

File: mix.exs [hunk 1]
```diff
@@ -10,26 +10,31 @@ defmodule PhoenixSlime.Mixfile do
...
-    [{:phoenix, "~> 1.3"},
-     {:phoenix_html, "~> 2.10"},
-     {:cowboy, "~> 1.0"},
-     {:slime, "~> 1.0"},
-     {:ex_doc, ">= 0.0.0", only: :dev},
-     {:earmark, ">= 0.0.0", only: :dev}]
+    [
+      {:phoenix, "~> 1.4"},
+      {:phoenix_html, "~> 2.10"},
+      {:jason, "~> 1.0", optional: true},
```
Comment (Rakoth) [inline mix.exs:25]: Why do we need jason in dependencies?

**Review by Rakoth — COMMENTED** [review]
Nice one, @scarfacedeb !

File: mix.exs (hunk 1 above)
Comment (scarfacedeb) [reply mix.exs:25]: To get rid of the following Phoenix warning:

```
~/t/phoenix_slime ❯❯❯ mix test
Compiling 5 files (.ex)
Generated phoenix_slime app
warning: failed to load Poison for Phoenix JSON encoding
(module Poison is not available).

Ensure Poison exists in your deps in mix.exs,
and you have configured Phoenix to use it for JSON encoding by
verifying the following exists in your config/config.exs:

    config :phoenix, :json_library, Poison


  (phoenix) lib/phoenix.ex:40: Phoenix.start/2
  (kernel) application_master.erl:277: :application_master.start_it_old/4

...........

Finished in 1.1 seconds
11 tests, 0 failures
```

Comment (Rakoth) [reply mix.exs:25]: Shouldn't it be only for test env then?

Comment (mschae) [reply mix.exs:25]: I believe `optional: true` is the way to go there...

Comment (scarfacedeb) [reply mix.exs:25]: I thought about another issue with the current approach:

Now we have implicit configuration (`config :phoenix, :json_library, Jason`) that sets config value that is unrelated to `phoenix_slime` and may break someone else's code. 

Therefore I think it's better to move the dependency into `:test` env and config into `config/test.exs`.

@Rakoth @mschae What do you think?

**Review by Rakoth — APPROVED** [review]

**Comment by Rakoth** [comment]
@doomspork I think, it is good to merge!

**Comment by lessless** [comment]
@doomspork do the needed please :)

**Comment by navinpeiris** [comment]
Any chance of getting a new release with this merge as well please? Thanks for all your work peeps! ❤️

**Comment by scarfacedeb** [comment]
@doomspork Sure, I'll be glad to help! 

I'm building API phoenix apps for the most part, but I have a couple of personal projects with slim templates too.

**Comment by doomspork** [comment]
@navinpeiris done!  Sorry for the delays, lots on my plate these days 🙁

**Comment by navinpeiris** [comment]
completely understand @doomspork, thanks so much! ❤️

**Pull request merged**